snuba/migrations/snuba_migrations/outcomes/0004_outcomes_matview_additions.py

```diff
@@ -0,0 +1,96 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt, String, DateTime
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+old_materialized_view_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("times_seen", UInt(64)),
+]
+
+new_materialized_view_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("category", UInt(8)),
+    Column("quantity", UInt(64, Modifiers(nullable=True))),
+    Column("times_seen", UInt(64)),
+]
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Updates materialized view query to support category and quantity
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.DropTable(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_mv_hourly_local",
+            ),
+            operations.CreateMaterializedView(
+                storage_set=StorageSetKey.OUTCOMES,
+                view_name="outcomes_mv_hourly_local",
+                destination_table_name="outcomes_hourly_local",
+                columns=new_materialized_view_columns,
+                query="""
+                    SELECT
+                        org_id,
+                        project_id,
+                        ifNull(key_id, 0) AS key_id,
+                        toStartOfHour(timestamp) AS timestamp,
+                        outcome,
+                        ifNull(reason, 'none') AS reason,
+                        category,
+                        count() AS times_seen,
+                        sum(quantity) AS quantity
```

Comment:
How would you use the quantity field for old data ?

Reply:
for old data, it won't be used as quantity doesn't have a meaning for ERROR data categories.